server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1038,7 +1061,11 @@ private DefaultSearchContext createSearchContext(ReaderContext reader, ShardSear
                 clusterService.state().nodes().getMinNodeVersion(),
                 validate,
                 indexSearcherExecutor,
-                this::aggReduceContextBuilder
+                this::aggReduceContextBuilder,
+                reader.indexService()
```

Comment:
You don't need this additional parameter, the `DefaultSearchContext` has access to `IndexService` and should be able to query the setting any time (it also has access to cluster settings via ``ClusterService`).

Reply:
> UT context. Is this a known limitation of clusterService in UTs?

By UT you mean integration tests or locally / remotely deployed environment?

Reply:
@jed326 You need to use `clusterService.getClusterSettings()` instead of `clusterService.getSettings()`